editor.planx.uk/src/pages/FlowEditor/components/Team/tests/TeamMembers.updateEditor.test.tsx

```diff
@@ -0,0 +1,187 @@
+import { screen, waitFor, within } from "@testing-library/react";
+import { useStore } from "pages/FlowEditor/lib/store";
+import { vi } from "vitest";
+
+import { setupTeamMembersScreen } from "./helpers/setupTeamMembersScreen";
+import { mockTeamMembersData } from "./mocks/mockTeamMembersData";
+import { mockPlainUser, mockPlatformAdminUser } from "./mocks/mockUsers";
+
+vi.mock(
+  "pages/FlowEditor/components/Team/queries/createAndAddUserToTeam.tsx",
+  () => ({
+    updateTeamMember: vi.fn().mockResolvedValue({
+      id: 1,
+    }),
+  }),
+);
+
+describe("when a user presses 'edit button'", () => {
+  beforeEach(async () => {
+    useStore.setState({
+      teamMembers: mockTeamMembersData,
+      user: mockPlatformAdminUser,
+    });
+
+    const { user } = await setupTeamMembersScreen();
+
+    const teamEditorsTable = screen.getByTestId("team-editors");
+    const addEditorButton = await within(teamEditorsTable).findByTestId(
+      "edit-button-0",
+    );
+
+    user.click(addEditorButton);
+    // Start each test with an open modal
+  });
+
+  it("opens the modal and displays the input fields with all inputs prepopulated", async () => {
+    expect(await screen.findByTestId("modal-edit-user")).toBeVisible();
+    const firstNameInput = await screen.findByLabelText("First name");
+    const lastNameInput = await screen.findByLabelText("Last name");
+    const emailInput = await screen.findByLabelText("Email address");
+    // 29 - 159
+    // Sorted based on first letter of first name Bill > Donella in Mocks
+    expect(firstNameInput).toHaveDisplayValue(mockTeamMembersData[1].firstName);
```

Comment:
I think the test would be easier to read if you use the string value of the name (and email etc) here, e.g. 
`expect(firstNameInput).toHaveDisplayValue("Donella");`
Instead of the next dev needing to click to look at the mockTeamMembersData file. I prefer the tests to tell the story as easily as possible :)